docs/fundamentals-and-concepts/stellar-data-structures/contracts.mdx

```diff
@@ -0,0 +1,58 @@
+---
+title: Smart Contracts
+sidebar_position: 50
+---
+
+:::note
+
+Stellar has integrated a smart contracts platform called "[Soroban](https://soroban.stellar.org/docs)" into the core protocol. This platform has now reached a stable release, and will be available on the Stellar network as of protocol v20.
+
+:::
+
+A smart contract is a programmed set of executable code and state that can be invoked or used on the Stellar network.
+
+## Wasm Bytecode
+
+Once a smart contract has been written by a developer and compiled into a WebAssembly (Wasm) executable file, it can then be "installed" onto the Stellar network. A `CONTRACT_DATA` [ledger entry](ledgers.mdx) is created to store this binary data and its unique identifier is the hash of the executable file. This binary executable is stored independently from its deployed contract(s). When a Stellar transaction attempts to invoke a contract function, the first thing to take place is this Wasm bytecode is retrieved from the ledger and a secure, isolated runtime virtual machine ("VM") is instantiated so it can run the bytecode for the contract and then exit.
```

Comment:
```suggestion
Once a smart contract has been written by a developer and compiled into a WebAssembly (Wasm) executable file, it can then be "installed" onto the Stellar network. A `CONTRACT_DATA` [ledger entry](ledgers.mdx) is created to store this binary data and its unique identifier is the hash of the executable file. This binary executable is stored independently from its deployed contract(s). When a Stellar transaction attempts to invoke a contract function, the  Wasm bytecode is first retrieved from the ledger and a secure, isolated runtime virtual machine ("VM") is instantiated so it can run the bytecode for the contract and then exit.
```